### PR TITLE
Bring back the changes of reading the version from FIRCore with fixes + unit test.

### PR DESCRIPTION
**Pull request by tejasd**
#no-changelog

### CODE REVIEW

File: Crashlytics/generate_project.sh [hunk 1]
```diff
@@ -19,4 +19,4 @@ readonly DIR="$( git rev-parse --show-toplevel )"
 
 "$DIR/Crashlytics/ProtoSupport/generate_crashlytics_protos.sh" || echo "Something went wrong generating protos.";
 
-pod gen "${DIR}/FirebaseCrashlytics.podspec" --auto-open --gen-directory="${DIR}/gen" --local-sources="${DIR}" --platforms=ios,macos,tvos,watchos --clean
+pod gen "${DIR}/FirebaseCrashlytics.podspec" --auto-open --gen-directory="${DIR}/gen" --local-sources="${DIR}" --platforms=ios,macos,tvos --clean
```
Comment (samedson) [inline Crashlytics/generate_project.sh:22]: Can you revert this change?

**Review by samedson — APPROVED** [review]
Just one thing 👍

File: Crashlytics/generate_project.sh (hunk 1 above)
Comment (tejasd) [reply Crashlytics/generate_project.sh]: Reverted! I need this for generate_project.sh to run locally - I haven't looked deeper into any setup I might've missed, but is that a known issue?

Comment (samedson) [reply Crashlytics/generate_project.sh]: Yeah I think there's some weird interaction with the podspec
